drivers/sn32/matrix_sn32f248b.c

```diff
@@ -346,22 +341,24 @@ uint8_t matrix_scan(void) {
     return matrix_changed;
 }
 
-/**
- * @brief   CT16B1 interrupt handler.
- *
- * @isr
- */
-OSAL_IRQ_HANDLER(SN32_CT16B1_HANDLER) {
-
-    chSysDisable();
+void rgb_callback(PWMDriver *pwmp);
 
-    OSAL_IRQ_PROLOGUE();
+void shared_matrix_enable(void) {
+    pwmcfg.callback = rgb_callback;
+    pwmEnablePeriodicNotification(&PWMD1);
+}
 
-    // Disable PWM outputs on column pins
-    SN_CT16B1->PWMIOENB = 0;
+void shared_matrix_disable(void) {
+    pwmcfg.callback = NULL;
+    pwmDisablePeriodicNotification(&PWMD1);
+}
 
-    SN_CT16B1->IC = mskCT16_MR24IC; // Clear match interrupt status
+void rgb_callback(PWMDriver *pwmp) {
 
+    // Disable PWM outputs on column pins
+    for(uint8_t i=0;i<24;i++){
+        pwm_lld_disable_channel(&PWMD1,i);
```

Comment:
Why is pwm_lld_disable_channel() used instead of pwmDisableChannel()? Isn't pwmDisableChannel() is the public one and pwm_lld_disable_channel() an internal function.

Reply:
resolved https://github.com/SonixQMK/qmk_firmware/pull/143/commits/8352be2de2597d5030400eb3bbe96505c7bea70f